### PR TITLE
Implement Motor Position Control

### LINKED ISSUE #22: Convert Motors to Position Control
Motors were initially made to be torque controlled. Since the Arduino is going to implement a PID controller, we can abstract it out to position control.

### PR DESCRIPTION
**Pull request by agupta231**
Closes #22.

Convert the motors into position control instead of torque controlled as this is how it will probably be interfaced with irl. Note that I also deleted one of the example scripts and replaced it with `gym_ solo/examples/solo8_vanilla/interactive_pos_control.py`, where the user can control the robot joint angles via literally dragging a slider on the screen.

### CODE REVIEW

File: gym_solo/envs/solo8v2vanilla.py [hunk 1]
```diff
@@ -23,6 +23,20 @@
 @dataclass
 class Solo8VanillaConfig(Solo8BaseConfig):
   urdf_path: str = 'assets/solo8v2/solo.urdf'
+  starting_joint_pos = {
```
Comment (mahajanrevant) [inline gym_solo/envs/solo8v2vanilla.py:26]: How did you come up with these values?

File: gym_solo/envs/solo8v2vanilla.py [hunk 2]
```diff
@@ -95,15 +108,21 @@ def reset(self, init_call: bool = False) -> solo_types.obs:
...
+    self.robot, joint_cnt = self._load_robot()
+    
+    joint_ordering = [self.client.getJointInfo(self.robot, j)[1].decode('UTF-8')
+                      for j in range(joint_cnt)]
+    positions = [self._config.starting_joint_pos[j] for j in joint_ordering]
+
+    # Let the robot lay down flat, as intended. Note that this is a hack
+    # around modifying the URDF, but that should really be handled in the
+    # solidworks phase
+    for i in range(500):
```
Comment (mahajanrevant) [inline gym_solo/envs/solo8v2vanilla.py:120]: Why did you change from 1000 to 500. Also what are these numbers

File: examples/solo8_vanilla/interactive_pos_control.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+from gym_solo.envs import solo8v2vanilla
+from gym_solo.core import obs
+from gym_solo.core import rewards
+from gym_solo.core import termination as terms
+
+
+if __name__ == '__main__':
+  config = solo8v2vanilla.Solo8VanillaConfig()
+  env: solo8v2vanilla.Solo8VanillaEnv = gym.make('solo8vanilla-v0', use_gui=True, realtime=True, config=config)
```
[the remark this answers is not in the document]
Comment (agupta231) [reply examples/solo8_vanilla/interactive_pos_control.py:16]: a variable `env` of type `solo8v2vanilla.Solo8VanillaEnv`

File: examples/solo8_vanilla/interactive_pos_control.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  env.obs_factory.register_observation(obs.TorsoIMU(env.robot))
+  env.reward_factory.register_reward(1,rewards.UprightReward(env.robot))
+  env.termination_factory.register_termination(
+    terms.TimeBasedTermination(10000000))
```
[the remark this answers is not in the document]
Comment (agupta231) [reply examples/solo8_vanilla/interactive_pos_control.py:21]: yeah just made a magic number so that the demo wouldn't time out. Would you prefer that I added a comment or just made a `Termination` which never ends (which is probably the correct way to do it)?

File: gym_solo/envs/solo8v2vanilla.py (hunk 1 above)
Comment (agupta231) [reply gym_solo/envs/solo8v2vanilla.py:26]: Experimentally found. Keeping it like this cause it's a "config" and it unique to the specific model. The only way to get around this would be to modify the file in solidworks as the urdf can't easily be edited to reorient all of the joints.

File: gym_solo/envs/solo8v2vanilla.py (hunk 2 above)
Comment (agupta231) [reply gym_solo/envs/solo8v2vanilla.py:120]: 1. Realized that 500 gets the robot in the exact same starting position as 1000, so choosing the 500 cause it's just less time / computation. 

2. What numbers?

**Review by agupta231 — COMMENTED** [review]
Replies are attached.

File: examples/solo8_vanilla/interactive_pos_control.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mahajanrevant) [reply examples/solo8_vanilla/interactive_pos_control.py:16]: Oh lol! My bad

File: examples/solo8_vanilla/interactive_pos_control.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mahajanrevant) [reply examples/solo8_vanilla/interactive_pos_control.py:21]: We want the episode to end at some point right?

File: gym_solo/envs/solo8v2vanilla.py (hunk 1 above)
Comment (mahajanrevant) [reply gym_solo/envs/solo8v2vanilla.py:26]: Got it

File: gym_solo/envs/solo8v2vanilla.py (hunk 2 above)
Comment (mahajanrevant) [reply gym_solo/envs/solo8v2vanilla.py:120]: What does 500 and 1000 signify

File: examples/solo8_vanilla/interactive_pos_control.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (agupta231) [reply examples/solo8_vanilla/interactive_pos_control.py:21]: Mentioned in the other PRs, but this is taken care of in #37

File: gym_solo/envs/solo8v2vanilla.py (hunk 2 above)
Comment (agupta231) [reply gym_solo/envs/solo8v2vanilla.py:120]: Just the number of time steps to run to let gravity drop the robot. Basically just a hack to get the robot into a deterministic start position. The number themselves are just magic numbers--they were experimentally found to work.

Comment (mahajanrevant) [reply gym_solo/envs/solo8v2vanilla.py:120]: Got it

File: examples/solo8_vanilla/interactive_pos_control.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mahajanrevant) [reply examples/solo8_vanilla/interactive_pos_control.py:21]: Got it

**Review by mahajanrevant — APPROVED** [review]
Looks good
